src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,46 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        System.out.println(width + " -- " + length + " -- " + color);
+
+        if (this == object) {
+            return true;
+        }
+
+        if (null == object) {
+            return false;
+        }
+
+        if (object instanceof Rectangle) {
+            Rectangle rectangle = (Rectangle) object;
+            return checkNullOrEquals(width, rectangle.getWidth())
+                    &&
+                    checkNullOrEquals(length, rectangle.getLength())
+                    &&
+                    color == rectangle.getColor();
+        }
+        return false;
+    }
+
+    private boolean checkNullOrEquals(Integer a, Integer b) {
+        return (null == a && null == b || a.equals(b));
```

Comment:
what if only a == null, but b don't. What will happen when it will check the right part?

Reply:
Updated. Done